internal/constellation/applyinit.go

```diff
@@ -0,0 +1,264 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package constellation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/edgelesssys/constellation/v2/bootstrapper/initproto"
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"github.com/edgelesssys/constellation/v2/internal/grpc/grpclog"
+	grpcRetry "github.com/edgelesssys/constellation/v2/internal/grpc/retry"
+	"github.com/edgelesssys/constellation/v2/internal/kms/uri"
+	"github.com/edgelesssys/constellation/v2/internal/retry"
+	"github.com/edgelesssys/constellation/v2/internal/state"
+	"github.com/edgelesssys/constellation/v2/internal/versions"
+	"google.golang.org/grpc"
+)
+
+// InitPayload contains the configurable data for the init RPC.
+type InitPayload struct {
+	MasterSecret    uri.MasterSecret
+	MeasurementSalt []byte
+	K8sVersion      versions.ValidK8sVersion
+	ConformanceMode bool
+}
+
+// GrpcDialer dials a gRPC server.
+type GrpcDialer interface {
+	Dial(ctx context.Context, target string) (*grpc.ClientConn, error)
+}
+
+// Init performs the init RPC.
+func (a *Applier) Init(
+	ctx context.Context,
+	dialer GrpcDialer,
```

Comment:
As suggested already, we could move this dialer to be a private attribute of `Applier`.
Setting it up could be done in the Constructor of the struct.